app/javascript/react/components/ReviewForm.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import ErrorList from "./ErrorList";
+import _ from 'lodash'
+
+const ReviewForm = (props) => {
+  const [errors, setErrors] = useState({})
+  const clearState = {
+    rating: "",
+    body: ""
+  }
+  const [reviewRecord, setReviewRecord] = useState(clearState)
+
+  const handleInputChange = event => {
+    setReviewRecord ({
+      ...reviewRecord,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const validForSubmission = () => {
+    let submitErrors = {}
+    const requiredFields = ["body", "rating"]
+    requiredFields.forEach(field => {
+      if (reviewRecord[field].trim() === "") {
+        submitErrors = {
+          ...submitErrors,
+          [field]: "is blank"
+        }
+      }
+    })
+  
+    setErrors(submitErrors)
+    return _.isEmpty(submitErrors)
+  }
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    if (validForSubmission()) {
+      let formPayload = {
+        review: reviewRecord
+      }
+  
+      try {
+        const response = await fetch(`/api/v1/movies/${props.movie.id}/reviews`, {
+          credentials: "same-origin",
+          method: "POST",
+          headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+          },
+          body: JSON.stringify(formPayload)
+        })
+        if (!response.ok) {
+          const errorMessage = `${response.status} (${response.statusText})`
+          throw new Error(errorMessage)
+        }
+        const responseBody = await response.json()
+        if (responseBody.errors) {
```

Comment:
this is ok, but it's assuming that any `errors` that come back are because a user isn't signed in - here, we could check if the error is about a user not signed in. Another approach is to prevent a user from submitting a review if they're not signed in for good user experience - we can do this by not displaying the new review form, and instead a link/button to the sign in page with a message that they need to sign in